src/store/wallet/wallet-epics.ts

```diff
@@ -54,7 +55,26 @@ const loadTokenAssetsEpic = (action$: Observable<Action>) =>
           }
         )
       ).pipe(
-        map(({ data }) => loadTokenBalancesActions.success(data.balances)),
+        switchMap(({ data }) =>
+          forkJoin(
+            data.balances.map(async balance => {
+              try {
+                const bestTokenMetadata = await getTokenMetadata(balance.contract, balance.token_id);
```

Comment:
this also should be used for `Add Token` modal (`loadTokenMetadataActions.submit` action)